libs/island-ui/core/src/lib/Typography/Typography.tsx

```diff
@@ -47,41 +48,47 @@ export const Typography = ({
   links,
   paddingTop,
   paddingBottom,
+  paddingY,
   fontWeight,
-}: TypographyProps) => (
-  <Cmp
-    id={id}
-    className={cn(
-      variant ? styles[variant] : null,
-      color ? colors[color] : null,
-      fontWeight ? fontWeightStyles[fontWeight] : null,
-      {
-        [truncateStyle]: truncate,
-        [linksStyle]: links,
-        [defaultFontWeights[variant]]: variant && !fontWeight,
-      },
-      paddingBottom !== undefined &&
-        resolveResponsiveProp(
-          paddingBottom,
-          spacing.paddingBottomXs,
-          spacing.paddingBottomSm,
-          spacing.paddingBottomMd,
-          spacing.paddingBottomLg,
-          spacing.paddingBottomXl,
-        ),
-      paddingTop !== undefined &&
-        resolveResponsiveProp(
-          paddingTop,
-          spacing.paddingTopXs,
-          spacing.paddingTopSm,
-          spacing.paddingTopMd,
-          spacing.paddingTopLg,
-          spacing.paddingTopXl,
-        ),
-    )}
-  >
-    {children}
-  </Cmp>
-)
+}: TypographyProps) => {
+  const resolvedPaddingTop = paddingTop ?? paddingY
+  const resolvedPaddingBottom = paddingBottom ?? paddingY
+
+  return (
+    <Cmp
```

Comment:
maybe we should rethink this and use <Box component={Cmp}> then we can reuse box styles and remove unnecessary styles from typography.treat (mostly spacings) and it could be easier to add sugar-syntax later on if that is what we want